clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -686,6 +696,207 @@ check_function_markers()
         if (!run_checker(memrefs, false))
             return false;
     }
+    // Correctly handle signal arriving between a branch instruction and the function
+    // entry.
+    {
+        constexpr addr_t SIG_HANDLER_PC = 400;
+        constexpr addr_t SYSCALL_PC = 500;
+        constexpr size_t SYSCALL_SZ = 2;
+        constexpr size_t RETURN_SZ = 3;
+
+        std::vector<memref_t> memrefs = {
+            gen_instr(TID, 1),
+            gen_instr_type(TRACE_TYPE_INSTR_DIRECT_CALL, TID, CALL_PC, CALL_SZ),
+            gen_marker(TID, TRACE_MARKER_TYPE_KERNEL_EVENT, CALL_PC + CALL_SZ),
+
+            gen_instr(TID, SIG_HANDLER_PC),
+            // Return from the signal handler.
+            gen_instr_type(TRACE_TYPE_INSTR_RETURN, TID, SIG_HANDLER_PC + 1, RETURN_SZ),
+            // sigreturn.
```

Comment:
nit: Inconsistent capitalization; plus style guide requests capitalized comments: "Sigreturn." and below "Syscall xfer." and repeated again multiple times.